zerver/management/commands/send_zulip_updates.py

```diff
@@ -0,0 +1,134 @@
+import os
+import re
+from datetime import timedelta
+from typing import Any
+
+from django.conf import settings
+from django.db import transaction
+from django.db.models import Q
+from django.utils.timezone import now as timezone_now
+from typing_extensions import override
+
+from zerver.actions.message_send import (
+    do_send_messages,
+    internal_prep_private_message,
+    internal_prep_stream_message,
+)
+from zerver.lib.management import ZulipBaseCommand
+from zerver.models.realm_audit_logs import RealmAuditLog
+from zerver.models.realms import Realm
+from zerver.models.users import get_system_bot
+
+zulip_updates_file_path = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), "zulip_updates.md"
+)
+
+
+def get_latest_zulip_updates_level() -> int:
+    with open(zulip_updates_file_path) as zulip_updates:
+        content = zulip_updates.read()
+        match = re.search(r"\*\*Zulip updates level (\d+)\*\*", content)
+        assert match is not None
+        return int(match.group(1))
+
+
+def get_zulip_update_message_for_level(level: int) -> str:
+    with open(zulip_updates_file_path) as zulip_updates:
+        content = zulip_updates.read()
+        # TODO: Add a comment explaining algo in plain words.
+        current_level = f"**Zulip updates level {level}**"
+        previous_level = f"**Zulip updates level {level-1}**"
+        pattern = re.compile(
+            f"{re.escape(current_level)}(.*?)" + f"{re.escape(previous_level)}", re.DOTALL
+        )
+        match = re.search(pattern, content)
+        assert match is not None
+        return match.group(1).strip()
+
+
+class Command(ZulipBaseCommand):
+    help = """Script to send zulip updates to realms."""
+
+    @override
+    def handle(self, *args: Any, **options: str) -> None:
+        latest_zulip_updates_level = get_latest_zulip_updates_level()
+
+        realms = Realm.objects.filter(
+            Q(zulip_updates_level__isnull=True)
+            | Q(zulip_updates_level__lt=latest_zulip_updates_level),
+            deactivated=False,
+        )
```

Comment:
Let's add a comment noting why we're doing `deactivated=False` here -- if the realm is later reactivated, then we'll actually send the notices that it missed while it was inactive. I think that's correct behavior.

Reply:
Thought about this, Should we send bunch of notices when someone reactivate ? -- If the time gap between deactivation - reactivation is large (say ~1-2 year) that will cause a lot of notices...

I think, the behaviour here will depend on "what are the practical cases when someone reactivate their realm"

Reply:
I think we are OK with a bundle of notices. I expect us to send these like once a month, not once a day, so it really won't be that much.

Reply:
Comment added.
```
    # Filter out deactivated realms. When a realm is later reactivated,
    # send the notices it missed while it was deactivated.
```